### ISSUE TITLE
Cassandra Docker Not Connecting

### ISSUE THREAD

**Issue by spencerlepine** [spencerlepine/squidshop-ecommerce#10]
## Context
Node.js Express API is trying to use `express-cassandra` driver ORM to make queries on the Cassandra database.

## Expected Behaviour
Should be able to make `.save` to the Cassandra database.
```js
router.post('/upload', (req, res) => {
  const productDetails = req.body;
  const newId = uuid();

  const newProduct = new models.instance.Product({
    ...productDetails,
    id: newId,
    created: Date.now(),
  });

  return newProduct.saveAsync(newId)
    .then(() => {
      res.status(201);
      return res.json({
        message: 'Successfully added product',
        productId: newId,
      });
    })
    .catch((err) => {
      console.log(err);

      res.status(409).json({
        message: 'Unable to create product',
        error: JSON.stringify(err),
      });
    });
});
```

## Actual Behaviour
Cannot connect to Cassandra at all..

<img width="840" alt="Screen Shot 2022-08-01 at 11 15 03 AM" src="https://user-images.githubusercontent.com/60903378/182214882-b29c688b-60cd-4abb-a6f8-372d056fae99.png">

**Comment by spencerlepine** [comment]
https://stackoverflow.com/questions/40443617/init-script-for-cassandra-with-docker-compose

**Comment by spencerlepine** [comment]
https://community.datastax.com/questions/9729/unable-to-access-cqlsh.html

**Comment by spencerlepine** [comment]
```sh
connection error: ('Unable to connect to any servers', {'192.168.160.2': error(111, "Tried connecting to [('192.168.160.2', 9042)]. Last error: Connection refused")})
```

**Comment by spencerlepine** [comment]
[https://community.datastax.com/questions/9342/cassandra-docker-container-goes-to-exited-state-af.html](https://community.datastax.com/questions/9342/cassandra-docker-container-goes-to-exited-state-af.html)
<img width="800" alt="Screen Shot 2022-08-01 at 11 41 19 AM" src="https://user-images.githubusercontent.com/60903378/182220746-66933fdf-7e77-451b-9d6a-0188d028de4a.png">

Need to specify host IP address, becuase cassandra default IP is `127.0.0.1`
```yml
    command: /bin/bash -c "echo loading cassandra keyspace && cqlsh `cassandra_db` -u cassandra -p cassandra -f /schema.cql --debug"
```

**Comment by spencerlepine** [comment]
Now I connected, and have migration script running.

Now when I try to query the database to POST a record, it gives me this error back:
```
ArgumentError: 'localDataCenter' is not defined in Client options
```

**Comment by spencerlepine** [comment]
Unconfigured table products

**Comment by spencerlepine** [comment]
Lots of mismatches with the the keyspace cql command:
```sql
DROP KEYSPACE IF EXISTS mykeyspace;

CREATE KEYSPACE mykeyspace WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}  AND durable_writes = true;

USE mykeyspace;
```

Note: was using NetworkTopologyStrategy` which was breaking it, many random things mismatching the cassandra config and the node js driver config

**Comment by spencerlepine** [comment]
These values must match:
```sql
CREATE TABLE Product (
    id varchar PRIMARY KEY,
    title varchar,
    price float,
    description varchar,
    category varchar,
    image varchar,
    rating_rate float,
    rating_count int,
    created timestamp,
    updated timestamp,
);
```

```js
// src/models/ProductModel.js
/*
CREATE TABLE products (
    id varchar PRIMARY KEY,
    title varchar,
    price float,
    description varchar,
    category varchar,
    image varchar,
    rating_rate float,
    rating_count int,
    created timestamp,
    updated timestamp,
);
*/
module.exports = {
  fields: {
    id: {
      type: 'varchar',
      default: { $db_function: 'uuid()' },
    },
    title: { type: 'varchar', default: 'untitled' },
    price: 'float',
    description: { type: 'varchar', default: 'no description provided' },
    category: { type: 'text', default: 'misc' },
    image: 'varchar',
    rating_rate: 'float',
    rating_count: 'int',
    created: {
      type: 'timestamp',
      default: { $db_function: 'toTimestamp(now())' },
    },
    updated: {
      type: 'timestamp',
      default: { $db_function: 'toTimestamp(now())' },
    },
  },
  key: ['id'],
};
```


```js
// src/models/index.js
...
cassandra.loadSchema('Product', ProductSchema);

```

**Comment by spencerlepine** [comment]
<img width="761" alt="Screen Shot 2022-08-01 at 12 01 07 PM" src="https://user-images.githubusercontent.com/60903378/182224341-4f89ee1c-7e4e-4add-9d7e-1b4b39336394.png">

**Comment by spencerlepine** [comment]
<img width="710" alt="Screen Shot 2022-08-01 at 12 01 35 PM" src="https://user-images.githubusercontent.com/60903378/182224534-2b01ba25-3fa3-4c8a-a333-411d23563dd0.png">

**Comment by spencerlepine** [comment]
Fixed with #11
